tests/topotests/bgp-auth/test_bgp_auth.py

```diff
@@ -357,9 +357,10 @@ def check_neigh_state(router, peer, state, vrf=""):
                 "show bgp vrf {} neighbors {} json".format(vrf, peer)
             )
         neigh_output_json = json.loads(neigh_output)
-        if neigh_output_json[peer]["bgpState"] == state:
-            matched = True
-            break
+        if peer in neigh_output_json.keys():
```

Comment:
is this just a case of the doppelganger issue?  Doesn't this just skip testing for the peer if it's not in the ouptut?

Reply:
No, this fails the test if the peer has not yet tried to connect. It only passes if the peer is in the output and the state is correct. 
This is just a check for the case where the remote peer has not sent a connect before the check is done.